doc/reference/reference_lua/checks.rst

```diff
@@ -0,0 +1,516 @@
+..  checks-module:
+
+Module checks
+=============
+
+**Since:** :doc:`2.11.0 </release/2.11.0>`
+
+The ``checks`` module provides the ability to check the types of arguments passed to a Lua function.
+You need to call the :ref:`checks(type_1, ...) <checks-checks>` function inside the target Lua function and pass :ref:`one or more <checks_number_of_arguments>` type qualifiers to check the corresponding argument types.
+There are two types of type qualifiers:
+
+-   A :ref:`string type qualifier <checks_string_type_qualifier>` checks whether a function's argument conforms to the specified type.
+-   A :ref:`table type qualifier <checks_table_type_qualifiers>` checks whether the values of a table passed as an argument conform to the specified types.
+
+.. NOTE::
+
+    For earlier versions, you can install the ``checks`` module from the `Tarantool rocks repository <https://www.tarantool.io/en/download/rocks>`_.
+
+.. _checks_loading:
+
+Loading checks
+--------------
+
+The :ref:`checks API <checks_api_reference>` is available in a script without loading the module using ``require('checks')``.
```

Comment:
> I actually don't get how our devs decide which modules need require and which don't

Someone just decides that's `rawset(_G, 'checks', checks)` is good idea five years ago and then it's just not removed by anyone. Actually, in case of repo-installed `checks` you also don't need to require it more than once. Once you have a require in any of your scripts, it is available in any code called later. It's just that build-in modules are required inside the core Tarantool code, so the first require already had happened for any application developer.

Actually, I prefer calling `local checks = require('checks')` in any script, but, well, it is a fast that it is not actually needed.